app/models/user.rb

```diff
@@ -39,6 +41,28 @@ class User < ActiveRecord::Base
     self.password_confirmation ||= User.default_password
   end
 
+
+  ## NOTE: Validation methods, including this one, are called twice when
+  # making new user from the admin interface. While not a behavior we want,
+  # it doesn't currently pose a problem.
+  def validate_email_pattern
+    if (!defined?(email) || (email == nil) || !email.index('@'))
+      errors.add(:email, 'is required and should end in @schools.nyc.gov or another participating school address')
+      false
+    end
+
+    allowed_email_patterns = AllowedUserEmailMasks.where(active:true).pluck(:email_pattern)
+
+    index = email.index('@')
```

Comment:
You could do `domain = email.downcase.strip.split('@')[-1]` which will return a string even if `@` isn't present.

Downcase and strip help normalize your data.  You may want to add them to after_create and after_update on the AllowedUserEmailMask model too.

Reply:
good idea

Reply:
made a few safety fixes around these lines